tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs

```diff
@@ -44,315 +46,392 @@ private sealed class TestClass<T>
             public T[] C0;
         }
 
-        /*[MethodImpl(MethodImplOptions.NoInlining)]
-        static void TestFillAllSpan(Span<byte> span)
+        // Copying the result of a computation to Sink<T>.Instance is a way
+        // to prevent the jit from considering the computation dead and removing it.
+        private sealed class Sink<T>
         {
-            for (int i = 0; i < span.Length; ++i) {
-                span[i] = unchecked((byte)i);
-            }
+            public T Data;
+            public static Sink<T> Instance = new Sink<T>();
         }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        static void TestFillAllArray(byte[] data)
-        {
-            for (int i = 0; i < data.Length; ++i) {
-                data[i] = unchecked((byte)i);
-            }
-        }
+        // Use statics to smuggle some information from Main to Invoke when running tests
+        // from the command line.
+        static bool IsXunitInvocation = true; // xunit-perf leaves this true; command line Main sets to false
+        static int InnerIterationCount = 0;   // used to communicate iteration count from BenchmarkAttribute
+                                              // (xunit-perf exposes the same in static property Benchmark.InnerIterationCount)
+        static bool DoWarmUp; // Main sets this when calling a new benchmark routine
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        static void TestFillAllReverseSpan(Span<byte> span)
+
+        // Invoke routine to abstract away the difference between running under xunit-perf vs running from the
+        // command line.  Inner loop to be measured is taken as an Action<int>, and invoked passing the number
+        // of iterations that the inner loop should execute.
+        static void Invoke(Action<int> innerLoop, string nameFormat, params object[] nameArgs)
         {
-            for (int i = span.Length; --i >= 0;) {
-                span[i] = unchecked((byte)i);
+            if (IsXunitInvocation)
+            {
+                foreach (var iteration in Benchmark.Iterations)
+                    using (iteration.StartMeasurement())
+                        innerLoop((int)Benchmark.InnerIterationCount);
```

Comment:
This is interesting. How would we run the benchmarks now? None of the tests with "Benchmark" attributes have iteration.StartMeasurement(). I thought the methods with Benchmark attributes are the entry point.